homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -239,8 +239,12 @@ async def async_step_ssdp(self, discovery_info: DiscoveryInfoType) -> FlowResult
         # Synology NAS can broadcast on multiple IP addresses, since they can be connected to multiple ethernets.
         # The serial of the NAS is actually its MAC address.
 
+        await self.async_set_unique_id(discovered_mac)
         existing_entry = self._async_get_existing_entry(discovered_mac)
 
+        if not existing_entry:
+            self._abort_if_unique_id_configured()
```

Comment:
Just do this:
```suggestion
        self._abort_if_unique_id_configured({CONF_HOST: parsed_url.hostname})
```

And it will update if necessary and then abort the flow. This means you can remove line 248-259.

Reply:
unfortunately this is not working here, since during discovery the field `ATTR_UPNP_SERIAL` contains the mac address of the device and is used as unique id, but during creation of config entry the serial is readout from device and set as unique id.
This is done, because the device can have multiple MAC addresses, those it can not be determined, which mac is the correct one, but the serial is unique per device.

Reply:
Ok! Maybe we should include that in a comment (or maybe we have and I missed it). Merging now to pull into beta.

Reply:
There is already a comment at lines 239-240, but honestly not as detailed as maybe needed 🤔